ext/standard/var_unserializer.re

```diff
@@ -1201,6 +1201,113 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *full_name, *enum_name, *case_name;
+	zend_class_entry *ce;
+
+	if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	full_name = zend_string_init(str, len, 0);
+	int32_t colon_pos = -1;
+	for (uint32_t i = 0; i < ZSTR_LEN(full_name); ++i) {
+		if (ZSTR_VAL(full_name)[i] == ':') {
+			colon_pos = i;
+			break;
+		}
+	}
+	if (colon_pos == -1) {
+		zend_string_release_ex(full_name, 0);
+		php_error_docref(NULL, E_WARNING, "Invalid enum name '%s' (missing colon)", ZSTR_VAL(full_name));
+		return 0;
+	}
+
+	enum_name = zend_string_init_fast(ZSTR_VAL(full_name), -(colon_pos - len) - 1);
+	case_name = zend_string_init_fast(&ZSTR_VAL(full_name)[colon_pos + 1], len - colon_pos - 1);
+	zend_string_release_ex(full_name, 0);
+
+	//if (!zend_is_valid_class_name(enum_name)) {
+	//	zend_string_release_ex(enum_name, 0);
+	//	zend_string_release_ex(case_name, 0);
+	//	return 0;
+	//}
+
+	ce = zend_lookup_class(enum_name);
+	if (!ce) {
+		php_error_docref(NULL, E_WARNING, "Class '%s' not found", ZSTR_VAL(enum_name));
+		zend_string_release_ex(enum_name, 0);
+		zend_string_release_ex(case_name, 0);
+		return 0;
+	}
+	if (!(ce->ce_flags & ZEND_ACC_ENUM)) {
+		php_error_docref(NULL, E_WARNING, "Class '%s' is not an enum", ZSTR_VAL(enum_name));
+		zend_string_release_ex(enum_name, 0);
+		zend_string_release_ex(case_name, 0);
+		return 0;
+	}
+
+	if (EG(exception)) {
+		zend_string_release_ex(enum_name, 0);
+		zend_string_release_ex(case_name, 0);
+		return 0;
+	}
+
+	*p = YYCURSOR;
```

Comment:
You also need to skip over the `"` and `;` - see the `"s:" uiv ":" ["] 	{` handler

```
	YYCURSOR += 2;
	*p = YYCURSOR;
```

**This will fix Zend/tests/enum/spl-object-storage.phpt**

Reply:
It did indeed, thanks!